src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -584,7 +591,7 @@ DQMatMulGQ2iP::DQMatMulGQ2iP(Context::Ref ctx) {
 
         if (ov::element::i4 == matched_qweight->get_element_type() && qweight_shape.size() == 3 &&
             ov::element::f16 == matched_qcoeff->get_element_type() && qcoeff_shape.size() == 3 &&
-            act_shape.size() == 3 && act_shape[1] > 1 &&  // multi-token case
+            act_shape.size() == 3 && (act_shape[0] > 1 || act_shape[1] > 1) &&  // multi-token case
```

Comment:
so `act_shape[0] > 1 || act_shape[1] > 1` misses the case where they both `>1` what means a batched case that we don't support.

To simplify this if, i'd move out this multi-token check to a dedicated boolean like

```
const auto just_one = [](std::size_t a, std::size_t b) {
    return (a == 1 && b > 1) || (a > 1 && b == 1);
};
```

and then check like
```suggestion
            act_shape.size() == 3 && just_one(act_shape[0], act_shape[1]) &&
```

Reply:
Done